luda-editor/luda-editor-client/src/app/router.rs

```diff
@@ -0,0 +1,57 @@
+use namui::Entity;
+
+use super::{
+    events::RouterEvent,
+    sequence_list::SequenceList,
+    types::{Page, RouterContext},
+};
+
+pub struct Router {
+    page: Page,
+    context: RouterContext,
+}
+
+impl Entity for Router {
+    type Props = ();
+
+    fn update(&mut self, event: &dyn std::any::Any) {
+        if let Some(event) = event.downcast_ref::<RouterEvent>() {
+            match &event {
+                RouterEvent::PageChangeToEditorEvent(initializer) => {
+                    self.page = Page::Editor(initializer(&self.context));
+                }
+                RouterEvent::PageChangeToSequenceListEvent(initializer) => {
+                    self.page = Page::SequenceList(initializer(&self.context));
+                }
+            }
+        }
+        match &mut self.page {
+            Page::Editor(editor) => editor.update(event),
+            Page::SequenceList(sequence_list) => sequence_list.update(event),
+        }
+    }
+
+    fn render(&self, props: &Self::Props) -> namui::RenderingTree {
+        match &self.page {
+            Page::Editor(editor) => editor.render(props),
+            Page::SequenceList(sequence_list) => sequence_list.render(props),
+        }
```

Comment:
If all `page.render`s have the same prop, We can use page like `current_page: Option<namui::Entity<Props>>`. Then we don't need to use match.